crates/miden-objects/src/account/component/mod.rs

```diff
@@ -112,6 +114,59 @@ impl AccountComponent {
             .with_supported_types(template.metadata().supported_types().clone()))
     }
 
+    /// Creates an [`AccountComponent`] from a [`Package`].
+    ///
+    /// This method extracts the account component metadata from the package and creates
+    /// a component with the provided storage slots. The component will have default
+    /// supported types (RegularAccountImmutableCode) unless overridden.
```

Comment:
```suggestion
    /// supported types ([`AccountType::RegularAccountImmutableCode`]) unless overridden.
```

Nit: I'd always write type names as a doc link. If we rename/refactor them in the future, this will notify us to update the docs, else they become outdated/wrong.